### PR TITLE
build: Remove the CockroachDB operator from Rook

### LINKED ISSUE #6990: Remove CockroachDB storage provider from Rook due to lack of community support
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The CockroachDB storage provider has not had sufficient community or maintainer support to continue to release its operator as part of Rook. The proposal is to remove CockroachDB from Rook in the v1.6 release (March timeline).

See the full proposal [here](https://docs.google.com/document/d/1TSPfqX3D08lN5RjIf2wbQideIMOv09bXreRMUFzIycY/edit).

**Expected behavior:**
Storage providers in Rook must have community and maintainer support. This has been discussed for the past two Rook community meetings as seen in the [notes](https://docs.google.com/document/d/1exd8_IG6DkdvyA0eiTtL2z5K2Ra-y68VByUUgwP7I9A/edit). Also see the related expectations for storage providers now being documented in #6989.

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The cockroachDB operator has not had community support in Rook. Therefore, the time has come to deprecate and remove it. If the sources are still needed, there is always git history.

**Which issue is resolved by this Pull Request:**
Resolves #6990 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: build/codegen/codegen.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # Disables quote checks, which is needed because of the SED variable here.
 
 KUBE_CODE_GEN_VERSION="kubernetes-1.17.2"
-GROUP_VERSIONS="rook.io:v1 rook.io:v1alpha2 ceph.rook.io:v1 cockroachdb.rook.io:v1alpha1 nfs.rook.io:v1alpha1 cassandra.rook.io:v1alpha1 edgefs.rook.io:v1 yugabytedb.rook.io:v1alpha1"
+GROUP_VERSIONS="rook.io:v1 rook.io:v1alpha2 ceph.rook.io:v1 nfs.rook.io:v1alpha1 cassandra.rook.io:v1alpha1 edgefs.rook.io:v1 yugabytedb.rook.io:v1alpha1"
```
Comment (leseb) [inline build/codegen/codegen.sh:21]: Oh yes, speed up my codegen 🕺

File: README.md [hunk 2]
```diff
@@ -76,14 +76,16 @@ More details about API versioning and status in Kubernetes can be found on the K
...
+| CockroachDB | [CockroachDB](https://www.cockroachlabs.com/product/cockroachdb/) is a cloud-native SQL database for building global, scalable cloud services that survive disasters.                                                                                                                                                  | cockroachdb.rook.io/v1alpha1 | [Deprecated](https://github.com/rook/rook/issues/6990)*                        |
+| EdgeFS      | [EdgeFS](http://edgefs.io) is high-performance and fault-tolerant decentralized data fabric with access to object, file, NoSQL and block.                                                                                                                                                                              | edgefs.rook.io/v1            | [Deprecated](https://github.com/rook/rook/issues/5823#issuecomment-703834989)* |
+
+\* CockroachDB was removed from Rook in v1.6. See [Rook v1.5](https://rook.github.io/docs/rook/v1.5/) docs if still interested.
```
Comment (leseb) [inline README.md:88]: Should we point out the issue too?

**Review by leseb — APPROVED** [review]
LGTM, do we have a PR to update the website too?

File: README.md (hunk 2 above)
Comment (travisn) [reply README.md:88]: The issue is actually already linked from the "deprecated" link above next to CockroachDB in the table.

**Comment by BlaineEXE** [comment]
Do we want to duplicate the rook/rook repo as rook/cockroachdb to preserve the state in case work starts up again, or would we just expect them to start anew and reference this commit to get the files back?

**Comment by travisn** [comment]
> Do we want to duplicate the rook/rook repo as rook/cockroachdb to preserve the state in case work starts up again, or would we just expect them to start anew and reference this commit to get the files back?

If an operator is brought back, I would recommend we create the repo at that point and they can use the git history.

**Review by BlaineEXE — APPROVED** [review]

**Comment by BlaineEXE** [comment]
Looks like cassandra integration test failed. Probably can rerun with skipIntegrationTests then pull?

**Pull request merged**